### PR TITLE
[PF-24] Creates a SA account in janitor-test for accessing cloud resources

### PR DESCRIPTION
**Pull request by yonghaoy**

### CODE REVIEW

**Review by mikeflinn — APPROVED** [review]

File: janitor-test/sa.tf [hunk 1]
```diff
@@ -0,0 +1,24 @@
+locals {
+  folder_roles = [
+    "roles/resourcemanager.folderAdmin",
+    "roles/resourcemanager.projectCreator",
+    "roles/resourcemanager.projectDeleter",
```
Comment (wchamber) [inline janitor-test/sa.tf:5]: does it need projectDeleter if the janitor SA does the deletion?

**Review by wchamber — CHANGES_REQUESTED** [review]
Should this have an output for the service account to use in terraform-ap-deployments to store the SA key in vault?

File: janitor-test/sa.tf [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  ]
+}
+
+# The service account used to access cloud resources in test.
+resource "google_service_account" "app" {
```
Comment (wchamber) [inline janitor-test/sa.tf]: let's rename from "app"

File: janitor-test/sa.tf (hunk 1 above)
Comment (yonghaoy) [reply janitor-test/sa.tf:5]: We need to test cleanup already deleted resources. And the easiest ways is to create -> delete -> publish a message. 
That is what we did for other similar tests.

File: janitor-test/sa.tf (hunk 2 above)
Comment (yonghaoy) [reply janitor-test/sa.tf]: done

File: janitor-test/sa.tf (hunk 1 above)
Comment (wchamber) [reply janitor-test/sa.tf:5]: ah, good point, thanks!

**Review by wchamber — APPROVED** [review]

**Pull request merged**